### PR TITLE
[NPU] Add `IPEX_LLM_NPU_MTL` to enable support on mtl

### PR DESCRIPTION
**Pull request by plusbang**
## Description

Background: https://github.com/analytics-zoo/nano/issues/1785

### 2. User API changes

To run optimized models on MTL NPU, need to set following variable:
```bash
set IPEX_LLM_NPU_MTL=1
```

No change for LNL.

### 4. How to test?
- [x] Application test

### CODE REVIEW

File: docs/mddocs/Quickstart/npu_quickstart.md [hunk 1]
```diff
@@ -90,6 +90,9 @@ For `ipex-llm` NPU support, set the following environment variable with active `
 
 ```cmd
 set BIGDL_USE_NPU=1
```
Comment (rnwang04) [inline docs/mddocs/Quickstart/npu_quickstart.md:92]: Maybe we can rename to `IPEX_LLM_USE_NPU` later ? Now it's a little strange with other env vars.

File: python/llm/src/ipex_llm/transformers/npu_models/convert_mp.py [hunk 2]
```diff
@@ -32,6 +32,11 @@ def convert_forward(m, target_m, new_forward):
 
 def optimize_llm_pre(model: torch.nn.Module, qtype, mixed_precision,
                      quantization_group_size=0, load=False, max_prompt_len=512):
+    if os.environ.get("IPEX_LLM_NPU_MTL", "0") != "0":
```
Comment (rnwang04) [inline python/llm/src/ipex_llm/transformers/npu_models/convert_mp.py:35]: if os.environ.get("IPEX_LLM_NPU_MTL", "0") == "1":

**Review by rnwang04 — APPROVED** [review]
others LGTM.

File: docs/mddocs/Quickstart/npu_quickstart.md (hunk 1 above)
Comment (plusbang) [reply docs/mddocs/Quickstart/npu_quickstart.md:92]: > ?

Good catch! I think we could also verify whether it's still necessary later : )

Comment (plusbang) [reply docs/mddocs/Quickstart/npu_quickstart.md:92]: > Maybe we can rename to `IPEX_LLM_USE_NPU` later ? Now it's a little strange with other env vars.

Good catch! I think we could also verify whether this is still necessary later : )

File: python/llm/src/ipex_llm/transformers/npu_models/convert_mp.py (hunk 2 above)
Comment (plusbang) [reply python/llm/src/ipex_llm/transformers/npu_models/convert_mp.py]: > if os.environ.get("IPEX_LLM_NPU_MTL", "0") == "1":

Have updated.

**Pull request merged**